edgeless_node/src/agent/mod.rs

```diff
@@ -65,33 +67,104 @@ impl Agent {
         // value: provider_id
         let mut resource_instances = std::collections::HashMap::<edgeless_api::function_instance::ComponentId, String>::new();
 
+        // After spawning a new function, the function´s class is only used to determine which runner to start it on.
+        // When stopping, only the stop_function_id is provided which does not allow to know which runner it is
+        // currently deployed on. Here, we implement a instance_id -> function_class HashMap
+        let mut component_id_to_class_map = std::collections::HashMap::<edgeless_api::function_instance::ComponentId, String>::new();
+
         log::info!("Starting Edgeless Agent");
         while let Some(req) = receiver.next().await {
             match req {
                 AgentRequest::Spawn(spawn_req) => {
                     log::debug!("Agent Spawn {:?}", spawn_req);
-                    match runner.start(spawn_req).await {
-                        Ok(_) => {}
-                        Err(err) => {
-                            log::error!("Unhandled Start Error: {}", err);
+
+                    // Save function_class for further interaction.
+                    // We can assume that the Optional<instance_id> is present.
+                    component_id_to_class_map.insert(
+                        spawn_req.instance_id.clone().unwrap().function_id,
+                        spawn_req.code.function_class_type.clone(),
+                    );
+
+                    // Get runner for function_class of spawn_req
+                    match runner.get_mut(&spawn_req.code.function_class_type) {
+                        Some(r) => {
+                            // Forward the start request to the correct runner
+                            match r.start(spawn_req).await {
+                                Ok(_) => {}
+                                Err(err) => {
+                                    log::error!("Unhandled Start Error: {}", err);
+                                }
+                            }
+                        }
+                        None => {
+                            log::error!("Could not find runner for {}", spawn_req.code.function_class_type);
```

Comment:
This is probably only a warning given it is based on a wrong request by another component.